### ISSUE TITLE
`test/test-constants.mjs` fails if package name & repo dirname mismatch

### ISSUE THREAD

**Issue by DerekNonGeneric** [DerekNonGeneric/loader339#12]
Test: [`test/test-constants.mjs`](https://github.com/DerekNonGeneric/loader339/blob/master/test/test-constants.mjs)

---

**TL;DR:** I didn't mean to write so much, please text-to-speech it. :)

If this repo is cloned using the following code snippet, which changes the name of the destination folder, the test in question will fail.

```console
git clone git@github.com:DerekNonGeneric/loader339.git loader339-next
```

I don't think I'm the only person who frequently changes repo's destination directory when cloning and I'm surprised there haven't been any reports yet. I've been aware of this problem for a few weeks now and I'm not exactly sure when it began. It's possible that it was introduced when paths started being converted into their absolute forms as is seen here:

https://github.com/DerekNonGeneric/loader339/blob/69ef4ffb44e43d41ed486c15688c8766ca4a7993/lib/constants.mjs#L7-L14

It would be cool if we could increase test specificity to avoid false positives. An alternative would be to document the recommended method of cloning (w/o changing destdir). For some organizations (including @ampproject), a pre-requisite to membership is having 2FA enabled, which means that all cloning essentially needs to happen using SSH for verification purposes. This doesn't work for everyone, so I would prefer we didn't need to provide cloning instructions aside from a link to [which-remote-url-should-i-use document](https://help.github.com/en/github/using-git/which-remote-url-should-i-use), which might go in [doc/development](https://github.com/DerekNonGeneric/loader339/blob/master/doc/development.md).

I think we're looking for feature detection, so including Node.js's platform tests here might not be appropriate. Perhaps simply succeeded to import using the `'loader339/constants'` specifier is sufficient evidence to cover a true-positive case.

/cc @MylesBorins

**Comment by DerekNonGeneric** [comment]
If we do end up using a fixture for this, depending on the directory structure we go with, some of our decisions may provide insight to help move https://github.com/nodejs/modules/issues/472 further along too. I know the ball is currently in my court on that issue, but any insight would be appreciated!

/cc @jkrems

**Comment by DerekNonGeneric** [comment]
Oof really sorry about my sample snippet. The one I provided is a rare case that it doesn't fail.

What I should've provided was the following.

```
git clone git@github.com:DerekNonGeneric/loader339.git lorem339; cd lorem339; npm i; npm t;
```
